_studio/mfx_lib/shared/src/mfx_h264_encode_vaapi.cpp

```diff
@@ -204,12 +205,12 @@ mfxStatus SetRateControl(
     if(par.calcParam.maxKbps)
         rate_param->target_percentage = (unsigned int)(100.0 * (mfxF64)par.calcParam.targetKbps / (mfxF64)par.calcParam.maxKbps);
 
-#if defined(LINUX_TARGET_PLATFORM_BXTMIN) || defined(LINUX_TARGET_PLATFORM_BXT) || defined(LINUX_TARGET_PLATFORM_CFL)
-    // Activate frame tolerance sliding window mode
-    if (extOpt3->WinBRCSize) {
-        rate_param->rc_flags.bits.frame_tolerance_mode = 1;
+    if (isBrcSlidingWindowFrameToleranceMode) {
+        // Activate frame tolerance sliding window mode
+        if (extOpt3->WinBRCSize) {
+            rate_param->rc_flags.bits.frame_tolerance_mode = 1;
```

Comment:
AFAIK there are at least 3 modes of frame size tolerance. Isn't it better to introduce enum for these modes instead of using magic number "1"?

Reply:
Yep, I agree. Here is extract from VAAPI:
```
            /** Frame tolerance mode.
             *
             *  Indicates the tolerance the application has to variations in the frame size.
             *  For example, wireless display scenarios may require very steady bit rate to
             *  reduce buffering time. It affects the rate control algorithm used,
             *  but may or may not have an effect based on the combination of other BRC
             *  parameters.  Only valid when the driver reports support for
             *  #VAConfigAttribFrameSizeToleranceSupport.
             *
             *  equals 0    -- normal mode;
             *  equals 1    -- maps to sliding window;
             *  equals 2    -- maps to low delay mode;
             *  other       -- invalid.
             */
            uint32_t frame_tolerance_mode   : 2;

```